File: Bryan/40.c
#include <stdio.h>
int main()
{
	char st[1000500];
	int i = 0;
	int k = 0;
	for(k=1;k<1000000;k++)
	{
		if (i >= 1000001)
			break;
		char buf[15];
		sprintf(&buf,"%d",k);
		int j = 0;
		while(buf[j])
			st[i++]=buf[j++];
	}
	int prod=1;
	int c = 1;
	for(i=0;i<7;i++)
	{
		prod*=(st[c-1]-48);
		c*=10;
	}
	printf("%d\n",prod);
}

File: Bryan/40.py
st = ""
counter = 0
for i in range(1,1000001):
	if len(st) >= 1000000:
		break
	st += str(i)
	if counter%300==0:
		print(len(st))
	counter+=1
xs = list(st)
prod = 1
c = 1
for i in range(7):
	prod*=int(xs[c-1])
	c*=10
print prod
